### ISSUE TITLE
"tokenize.whitespace" should not deal with unicode whitespaces

### ISSUE THREAD

**Issue by carltonwang** [stanfordnlp/CoreNLP#537]
I have a pre-tokenized text, so I join the tokens by " ", and set the "tokenize.whitespace" to "true" to skip the tokenization. But since there are some unicode whitespaces (U+200B: \xe2\x80\x8b) in some tokens, the tokenizer still split each of these tokens into two or more tokens. How can I stop the tokenizer from doing this?

**Comment by manning** [comment]
It was always the intention that the whitespace tokenizer _should_ be aware of Unicode whitespace, such as Japanese ideographic space (U+3000). Nevertheless, this class hadn't been updated for a long time, and wasn't fully consistent with what we did elsewhere, e.g., for PTBTokenizer. 

I've just edited it to make it reasonably consistent with other things.  The class now almost perfectly follows the definitions of the Java [isWhitepace() method](http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Character.html#isWhitespace(int)). People argue about that one too, but it seems a reasonable precedent to follow. (It differs only in also allowing U+0085 as a line separator, which helps on certain IBM systems, and probably does no harm.)

In particular, if you want to have "spaces" in tokens, the way to have it work (now) is to make them non-breaking spaces (U+00A0). This is the same convention used in PTBTokenizer.
